MiBench/security/rijndael/src/aesxam.c

```diff
@@ -1,311 +1,102 @@
-    i = 0;          /* this is a count for the input digits processed   */
-    
-    while(i < 64 && *cp)    /* the maximum key length is 32 bytes and   */
-    {                       /* hence at most 64 hexadecimal digits      */
-        ch = toupper(*cp++);            /* process a hexadecimal digit  */
-        if(ch >= '0' && ch <= '9')
+    cp = argv[4]; /* this is a pointer to the hexadecimal key digits  */
+    i = 0;        /* this is a count for the input digits processed   */
+
+    while (i < 64 && *cp) /* the maximum key length is 32 bytes and   */
+    {                     /* hence at most 64 hexadecimal digits      */
+        ch = toupper(*cp++); /* process a hexadecimal digit  */
+        if (ch >= '0' && ch <= '9')
             by = (by << 4) + ch - '0';
-        else if(ch >= 'A' && ch <= 'F')
+        else if (ch >= 'A' && ch <= 'F')
             by = (by << 4) + ch - 'A' + 10;
-        else                            /* error if not hexadecimal     */
+        else /* error if not hexadecimal     */
         {
-            printf("key must be in hexadecimal notation\n"); 
-            err = -2; goto exit;
+            printf("key must be in hexadecimal notation\n");
+            err = -2;
+            goto exit;
         }
-        
+
         /* store a key byte for each pair of hexadecimal digits         */
-        if(i++ & 1) 
-            key[i / 2 - 1] = by & 0xff; 
+        if (i++ & 1)
+            key[i / 2 - 1] = by & 0xff;
     }
 
-    if(*cp)
+    if (*cp)
     {
-        printf("The key value is too long\n"); 
-        err = -3; goto exit;
+        printf("The key value is too long\n");
+        err = -3;
+        goto exit;
     }
-    else if(i < 32 || (i & 15))
+    else if (i < 32 || (i & 15))
     {
         printf("The key length must be 32, 48 or 64 hexadecimal digits\n");
-        err = -4; goto exit;
+        err = -4;
+        goto exit;
     }
 
     key_len = i / 2;
 
-    if(!(fin = fopen(argv[1], "rb")))   /* try to open the input file */
+    if (!(fin = fopen(argv[1], "rb"))) /* try to open the input file */
     {
-        printf("The input file: %s could not be opened\n", argv[1]); 
-        err = -5; goto exit;
+        printf("The input file: %s could not be opened\n", argv[1]);
+        err = -5;
+        goto exit;
     }
 
-    if(!(fout = fopen(argv[2], "wb")))  /* try to open the output file */
+    if (!(fout = fopen(argv[2], "wb"))) /* try to open the output file */
     {
-        printf("The output file: %s could not be opened\n", argv[1]); 
-        err = -6; goto exit;
+        printf("The output file: %s could not be opened\n", argv[1]);
+        err = -6;
+        goto exit;
     }
 
-    if(toupper(*argv[3]) == 'E')
-    {                           /* encryption in Cipher Block Chaining mode */
+    if (toupper(*argv[3]) == 'E')
+    { /* encryption in Cipher Block Chaining mode */
         set_key(key, key_len, enc, ctx);
```

Comment:
should this not be included?

Reply:
I am not sure, based on argv[3], it is choosing between the encryption or decryption